httpie/uploads.py

```diff
@@ -64,13 +67,53 @@ def wrapped(*args, **kwargs):
     return wrapped
 
 
+def is_stdin(file: IO) -> bool:
+    try:
+        file_no = file.fileno()
+    except Exception:
+        return False
+    else:
+        return file_no == sys.stdin.fileno()
+
+
+READ_THRESHOLD = float(os.getenv("HTTPIE_STDIN_READ_THRESHOLD", 10.0))
```

Comment:
> I wonder: Should make this configurable (--xxx=60) & disableable (--xxx=0)?

I am not sure if it would be useful at all, considering this behavior has no side-effect (e.g, we do not terminate the process), just a warning. I explicitly ignored to have this documented, since this is primarily a test-only env var (there is no way to mock stuff on tests where I need to create subprocesses). I'd prefer make it just an invisible configuration var, until someone official asks whether they can disable it or not.

> Also, single quotes (many places).

My bad :/ I'll integrate the [pre-commit-hooks/double-quote-fixer](https://github.com/pre-commit/pre-commit-hooks#double-quote-string-fixer) right after 3.0 release, so we won't need to worry about this.

Reply:
> I am not sure if it would be useful at all, considering this behavior has no side-effect

If you do run this from a `cron` job like `/bin/slow-script | http POST pie.dev/post`, then the warning will have a side-effect in the form of an email, for example.

Also, I believe the current implementation doesn’t respect [`-qq`](https://httpie.io/docs/cli/verbose-output).

Reply:
> If you do run this from a cron job like /bin/slow-script | http POST pie.dev/post, then the warning will have a side-effect in the form of an email, for example.

Am I missing something:
```console
$ python -c 'import time; time.sleep(2); print("regular data")' | HTTPIE_STDIN_READ_THRESHOLD=1 http -p=B POST pie.dev/post                                                                                                                                                        2,345s

http: warning: > no stdin data read in 1.0s (perhaps you want to --ignore-stdin)
> See: https://httpie.io/docs/cli/best-practices

regular data


```
This does not affect the request body.

Reply:
> Also, I believe the current implementation doesn’t respect -qq.

👍🏻 TIL. Will add support for this one.

Reply:
I’m referring to the behavior of `cron` where any output triggers an email being sent (an example of a side-effect the output can have):

``` 
$ man cron
```

> The cron utility then wakes up every minute, examining all stored
>      crontabs, checking each command to see if it should be run in the
>      current minute.  **When executing commands, any output is mailed to
>      the owner of the crontab** (or to the user named in the MAILTO
>      environment variable in the crontab, if such exists).
>

Reply:
> I’m referring to the behavior of cron where any output triggers an email being sent (an example of a side-effect the output can have — admittedly a confusing one):

Ah, that makes sense. Then the question becomes whether we should only support `-qq` or add an additional configuration option. I'd go with `-qq` (which I believe is already supported).

Reply:
Alright, let’s keep things simple and rely on `--qq` for now. We can revisit this if we receive some complaints about this warning. But we should probably handle `HTTPIE_STDIN_READ_THRESHOLD=0` despite that not being documented (also, this might be a slightly better name: `HTTPIE_STDIN_READ_WARN_THRESHOLD`).

Reply:
We could also prefix the name with an underscore to _underscore_ it’s an internal variable.

Reply:
> We could also prefix the name with an underscore to underscore it’s an internal variable.

The Python level name or the environment variable? I don't think I've ever seen an environment variable with an underscore prefix.